File: PO/Project/C++/src/wolfberry.cpp
#include "wolfberry.h"

wolfberry::wolfberry(YX position, world* world_point)
: plant(99, position ,world_point)
{
    avatar = 'w';
}

organizm* wolfberry::MakeChild(YX new_position)
{
    return (new wolfberry(new_position, world_point));
}

wolfberry::~wolfberry()
{
}

File: PO/Project/JavaProject/src/Plants/Wolfberry.java
package Plants;

import Abstractions.*;
import java.awt.*;

public class Wolfberry extends Plant {

    public Wolfberry(World world,Point position) {
        super(99, position, world);
        this.avatar = 'w';
        this.image_path = "Wolfberry.png";
        loadImage();
    }

    @Override
    public boolean rejectAttack(Organism attacker) {
        attacker.die();
        world.getLogger().log("Wolfberry instantly killed an attacker");
        return true;
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Wolfberry(world, newPosition);
    }
}
